integration/run_test.go

```diff
@@ -0,0 +1,159 @@
+	flag.Parse()
+	if *remote {
+		cmd := exec.Command("gcloud", "container", "clusters", "get-credentials", *gkeClusterName, "--zone", *gkeZone, "--project", *gcpProject)
+		if err := integration_util.RunCmd(cmd); err != nil {
+			logrus.Fatalf("Error authenticating to GKE cluster stdout: %v", err)
+		}
+	}
+
+	var err error
+	client, err = kubernetesutil.GetClientset()
+	if err != nil {
+		logrus.Fatalf("Test setup error: getting kubernetes client: %s", err)
+	}
+
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
+
+	cfg, err := kubeConfig.RawConfig()
+	if err != nil {
+		logrus.Fatalf("loading kubeconfig: %s", err)
+	}
+
+	context = cfg.Contexts[cfg.CurrentContext]
+
+	exitCode := m.Run()
+
+	// Reset default context and namespace
+	if err := exec.Command("kubectl", "config", "set-context", context.Cluster, "--namespace", context.Namespace).Run(); err != nil {
+		logrus.Warn(err)
+	}
+
+	os.Exit(exitCode)
+}
+
+func setupNamespace(t *testing.T) (*v1.Namespace, func()) {
+	namespaceName := integration_util.RandomID()
+	ns, err := client.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      namespaceName,
+			Namespace: namespaceName,
+		},
+	})
+	if err != nil {
+		t.Fatalf("creating namespace: %s", err)
+	}
+
+	kubectlCmd := exec.Command("kubectl", "config", "set-context", context.Cluster, "--namespace", ns.Name)
+	if err := integration_util.RunCmd(kubectlCmd); err != nil {
+		t.Fatalf("kubectl config set-context --namespace: %v", err)
+	}
+
+	os.Setenv("KRITIS_DEPLOY_NAMESPACE", namespaceName)
+
+	return ns, func() {
+		client.CoreV1().Namespaces().Delete(ns.Name, &meta_v1.DeleteOptions{})
+		os.Setenv("KRITIS_DEPLOY_NAMESPACE", "")
+	}
+}
+
+var CRDS = []string{
+	"attestation-authority-crd.yaml",
+	"image-security-policy-crd.yaml",
+}
+
+var CRD_EXAMPLES = []string{
+	"attestation-authority-example.yaml",
+	"image-security-policy-example.yaml",
+}
+
+func deleteCRDs() {
+	for _, crd := range CRDS {
+		crdCmd := exec.Command("kubectl", "delete", "-f",
+			crd)
+		crdCmd.Dir = "../artifacts/examples"
+		integration_util.RunCmdOut(crdCmd)
+	}
+}
+
+func createCRDs(t *testing.T) {
+	for _, crd := range CRDS {
+		crdCmd := exec.Command("kubectl", "create", "-f",
+			crd)
+		crdCmd.Dir = "../artifacts/examples"
+		_, err := integration_util.RunCmdOut(crdCmd)
+		if err != nil {
+			t.Fatalf("testing error: %v", err)
+		}
+	}
+}
+
+func createCRDExamples(t *testing.T) {
+	for _, crd := range CRD_EXAMPLES {
+		crdCmd := exec.Command("kubectl", "create", "-f",
+			crd)
+		crdCmd.Dir = "../artifacts/examples"
+		_, err := integration_util.RunCmdOut(crdCmd)
+		if err != nil {
+			t.Fatalf("testing error: %v", err)
+		}
+	}
+}
+
+func TestCRDs(t *testing.T) {
+	_, deleteNs := setupNamespace(t)
+	defer deleteNs()
+
+	// CRDs themselves are non-namespaced so we have to delete them each run
```

Comment:
Should you add another call to this, defered?

Reply:
yea, added